lib/jxl/dec_frame.cc

```diff
@@ -1040,11 +1039,11 @@ int FrameDecoder::References() const {
   if (is_finalized_) {
     return 0;
   }
-  if ((!decoded_dc_global_ || !decoded_ac_global_ ||
-       *std::min_element(decoded_dc_groups_.begin(),
-                         decoded_dc_groups_.end()) != 1 ||
-       *std::min_element(decoded_passes_per_ac_group_.begin(),
-                         decoded_passes_per_ac_group_.end()) < max_passes_)) {
+  if (!(decoded_dc_global_ && decoded_ac_global_ &&
+        *std::min_element(decoded_dc_groups_.begin(),
```

Comment:
I realize this is preexisting, but could we do std::any_of / std::all_of instead of min_element? it would be, I think, a lot clearer.

Reply:
Sure. Do you happen to know a more elegant way to do that than with a lambda predicate like this?

```
std::all_of(decoded_dc_groups_.begin(), decoded_dc_groups_.end(), [&](uint8_t i) { return i; })
```

Reply:
I wouldn't [&] ;)

Otherwise, I'd just use that lambda tbh - perhaps naming it somewhere so it's clear what it is.

Reply:
How about just `!std::find(decoded_dc_groups_.begin(), decoded_dc_groups_.end(), 0)` ?

Reply:
also works (but `uint8_t{false}`)

Reply:
nevermind, I was thinking `find` returns nullptr if it didn't find it, but it doesn't

Reply:
I did something else to make things more readable, PTAL.